src/components/JoinSwapExternAmount.svelte

```diff
@@ -58,10 +57,10 @@
   $: isFetchingQuote = false;
   $: error = null;
   $: showSlippageSettings = false;
-  $: balanceError = (sellToken && amount && sellToken?.balance && amount.bn.isGreaterThan(sellToken.balance.bn));
+  $: balanceError = (sellToken && sellAmount && sellToken?.balance && sellAmount.bn.isGreaterThan(sellToken.balance.bn));
   $: tokensLoaded = tokenList.length > 0;
-  
-  
+  $: sellTokenDecimals = sellToken?.decimals ?? 18;
```

Comment:
I think this could be included on the defaultAmount so you don't actually extend the default value for decimals here

Reply:
That's a very nice spot